tests/Microsoft.Identity.Test.Unit/ApiConfigTests/AuthorityTests.cs

```diff
@@ -62,25 +82,142 @@ public void WithTenantIdExceptions()
+        [DataRow(TestConstants.CiamAuthorityWithFriendlyName)]
+        [DataRow(TestConstants.CiamAuthorityWithGuid)]
+        public void WithTenantId_NonAad(string inputAuthority)
+        {
+            var app = ConfidentialClientApplicationBuilder
                .Create(TestConstants.ClientId)
-               .WithB2CAuthority(TestConstants.B2CAuthority)
+               .WithAuthority(inputAuthority)
                .WithClientSecret("secret")
                .Build();
 
-            var ex2 = AssertException.Throws<MsalClientException>(() =>
-                app2
-                    .AcquireTokenByAuthorizationCode(TestConstants.s_scope, "code")
-                    .WithTenantId(TestConstants.TenantId));
+            var parameterBuilder = app
+                .AcquireTokenByAuthorizationCode(TestConstants.s_scope, "code")
+                .WithTenantId(TestConstants.TenantId2);
+
+            Assert.AreEqual(
+                new Uri(inputAuthority).Host, 
+                parameterBuilder.CommonParameters.AuthorityOverride.Host, 
+                "The host should have stayed the same");
+            
+            Assert.AreEqual(
+                TestConstants.TenantId2,
+                AuthorityHelpers.GetTenantId(parameterBuilder.CommonParameters.AuthorityOverride.CanonicalAuthority),
+                "The tenant id should have been changed");
+        }
 
-            Assert.AreEqual(ex1.ErrorCode, MsalError.TenantOverrideNonAad);
-            Assert.AreEqual(ex2.ErrorCode, MsalError.TenantOverrideNonAad);
+        [DataTestMethod]
+        [DataRow(TestConstants.AuthorityCommonTenant)]
+        [DataRow(TestConstants.AuthorityCommonPpeAuthority)]
+        [DataRow(TestConstants.AuthorityConsumersTenant)]
+        [DataRow(TestConstants.AuthorityOrganizationsTenant)]
+        [DataRow(TestConstants.AuthorityGuidTenant)]
+        [DataRow(TestConstants.DstsAuthorityCommon)]
+        [DataRow(TestConstants.DstsAuthorityTenanted)]
+        [DataRow(TestConstants.CiamAuthorityMainFormat)]
+        [DataRow(TestConstants.CiamAuthorityWithFriendlyName)]
+        [DataRow(TestConstants.CiamAuthorityWithGuid)]
+        public void AppLevel_AuthorityAndTenant_TenantWins_Config(string inputAuthority)
+        {
+            var cca = ConfidentialClientApplicationBuilder
+                .Create(TestConstants.ClientId)
+                .WithTenantId(TestConstants.TenantId2)
+                .WithAuthority(inputAuthority)
+            .Build();
+
+            Assert.AreEqual(
+                new Uri(inputAuthority).Host,
+                (cca.AppConfig as ApplicationConfiguration).Authority.AuthorityInfo.Host,
+                "The host should have stayed the same");
+
+            Assert.AreEqual(
+                TestConstants.TenantId2, 
+                AuthorityHelpers.GetTenantId(new Uri(cca.Authority)));
+        }
+
+        [TestMethod]
+        public void B2CAuthorityWithTenant()
```

Comment:
This name and the name of the test on 195 are a bit too similar, should propably add `AppLevel...` and `RequestLevel...` to differentiate. Maybe even group them.